crates/bevy_ecs/src/lib.rs

```diff
@@ -721,65 +693,73 @@ mod tests {
         {
             world
                 .query_filtered::<Entity, F>()
-                .iter(&world)
+                .iter()
                 .collect::<Vec<Entity>>()
         }
 
-        assert_eq!(get_filtered::<Mutated<A>>(&mut world), vec![e1, e3]);
+        assert_eq!(
+            get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world),
+            vec![e1, e3]
+        );
 
         // ensure changing an entity's archetypes also moves its mutated state
         world.entity_mut(e1).insert(C);
 
-        assert_eq!(get_filtered::<Mutated<A>>(&mut world), vec![e3, e1], "changed entities list should not change (although the order will due to archetype moves)");
+        assert_eq!(get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world), vec![e3, e1], "changed entities list should not change (although the order will due to archetype moves)");
 
         // spawning a new A entity should not change existing mutated state
         world.entity_mut(e1).insert_bundle((A(0), B));
         assert_eq!(
-            get_filtered::<Mutated<A>>(&mut world),
+            get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world),
             vec![e3, e1],
             "changed entities list should not change"
         );
 
         // removing an unchanged entity should not change mutated state
         assert!(world.despawn(e2));
         assert_eq!(
-            get_filtered::<Mutated<A>>(&mut world),
+            get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world),
             vec![e3, e1],
             "changed entities list should not change"
         );
 
         // removing a changed entity should remove it from enumeration
         assert!(world.despawn(e1));
         assert_eq!(
-            get_filtered::<Mutated<A>>(&mut world),
+            get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world),
             vec![e3],
             "e1 should no longer be returned"
         );
 
         world.clear_trackers();
+        world.exclusive_system_counter = world.increment_global_system_counter();
 
-        assert!(get_filtered::<Mutated<A>>(&mut world).is_empty());
+        assert!(get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world).is_empty());
 
         let e4 = world.spawn().id();
 
         world.entity_mut(e4).insert(A(0));
-        assert!(get_filtered::<Mutated<A>>(&mut world).is_empty());
+        assert!(get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world).is_empty());
         assert_eq!(get_filtered::<Added<A>>(&mut world), vec![e4]);
 
         world.entity_mut(e4).insert(A(1));
-        assert_eq!(get_filtered::<Mutated<A>>(&mut world), vec![e4]);
+        // assert_eq!(get_filtered::<(Changed<A>, Not<Added<A>>)>(&mut world), vec![e4]); // This case is no longer possible to detect
```

Comment:
The difference is that this case used to set both the added and mutated flag at the same time. It would only be possible in exclusive systems, where you insert and then modify a component. It's the only case where `Mutated<T>` is not equivalent to `(Changed<T>, Not<Added<T>>)`.
It would be possible to get that functionality back by checking if the added and changed counters are equal, but I don't think it's useful.

Reply:
Great. Yeah, I can't see any particular use for this: I think we should just remove the line completely.

Reply:
Agreed, I left it to make sure it would be discussed during review. Removed in 7d38cc4426abe99143cc479bf5befb63de8020c0